core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -2011,6 +2006,38 @@ public static boolean equalType(String desc0, MutableRel rel0, String desc1,
     return RelOptUtil.equal(desc0, rel0.rowType, desc1, rel1.rowType, litmus);
   }
 
+  /**
+   * Check filter bottom join can be pull-up,
```

Comment:
Check if filter under join can be pulled up

Reply:
ok, thanks for your remind.